rpc/client.go

```diff
@@ -768,7 +776,7 @@ func (sub *ClientSubscription) forward() (err error, unsubscribeServer bool) {
 			if err != nil {
 				return err, true
 			}
-			if buffer.Len() == maxClientSubscriptionBuffer {
+			if buffer.Len() == sub.maxClientSubscriptionBuffer {
 				return ErrSubscriptionQueueOverflow, true
```

Comment:
Ultimately fixes outside `swarm/pss` should go in seperate branches, I believe. Please make a note of them, so they can be cherry-picked to another branch down the line. (agree @lmars ?)

Reply:
Yes please keep them in a separate commit.

Reply:
Sure, will keep them separate. I need to do a bit more investigation though with respect to this buffer, and understand why we filling it up. I am not sure the best solution is to just increase it, let alone include a size with every single subscription.

Reply:
How was this again; the buffer problem is exclusive to `pss`, or did it appear in your simpler tests, too? If the latter, I would advise creating a new thread off `swarm-network-rewrite` and fix the problem with minimum added complexity.

Reply:
@nolash 
1) the client subscription buffer problem was visible on my simpler tests as well, as soon as I went above 8000 subscriptions.
2) the unix socket buffer problem is visible only on the pss_test.go, probably due to the more even read/write flow on the simpler tests, not allowing the buffer to fill up. On `pss_test.go` when adding metrics, it is noticable that we issue more writes, than reads, which explains the buffer filling up and eventually blocking the writes.

![screen shot 2017-10-12 at 14 09 27](https://user-images.githubusercontent.com/50459/31654973-d57c7f38-b327-11e7-8eb9-ade42ae6121c.png)

Reply:
Which would be perfectly logical because it takes a bit of time to route the messages, right (nodes pass on, maybe even try to decrypt the messages).

I'm a bit worried that we are creating artificial scenarios that the software isn't intended to fulfill. Is this barrage of writes without any delay even realistic? And if it is, maybe we should filter them before the send calls?

Of course, I appreciate the importance of theoretical tests to gauge the capacity of the simulation framework. However, I'm unsure if it's 1) fair or 2) advisable to be using `pss` for it. The should be made in an enviroment that's as controllable as possible. So I make the case of reverting to generic tests, and build complexity and overhead on them until we make them fail.

Reply:
Your questions are valid, and I also had the feeling that at times I am abusing the system with the stress tests. Ideally I want to build a bit more intuition and answer those questions properly before deciding if the scenarios are valid or not, I don't have conclusive answers so far. (this is why I am reluctant to increase buffers just for the sake of tests at this point).

I believe with a bit more investigation, we can get to the bottom of it, understand the asynchronous behaviour of PSS through the sim adapter, and decide what makes most sense from all these tests.